### ISSUE TITLE
Excessive GPU memory usage at finetune stage

### ISSUE THREAD

**Issue by chu-tianxiang** [chu-tianxiang/QuIP#7]
Currently the end2end finetune stage consumes more GPU memory than expected, causing OOM when quantizing large models.

Possible reasons include:
* gradient checkpointing is not enabled.
* finetuning embedding and output layers in fp32 precision needs too much memory.

**Comment by chu-tianxiang** [comment]
The extra memory problem has been solved in [this commit](https://github.com/chu-tianxiang/QuIP-for-all/commit/e81125ef77f9cf76a5285c1ea9a2be3f277945cf) and now it supports using multiple GPUs for end2end finetune.

I'm running test with Qwen1.5-72B using 2 GPUs nows. With `ft_embedding=False`,  the maximum memory reported by `torch.cuda.max_memory_allocated` is 23.5G.

**Comment by Rain19981998** [comment]
What device is used to quantify the qwen-72B model，8*A100-40G can run it？

**Comment by chu-tianxiang** [comment]
> What device is used to quantify the qwen-72B model，8*A100-40G can run it？

Two A100-40G should suffice.
